src/software/new_geom/spline2d.h

```diff
@@ -89,6 +91,16 @@ class Spline2d
      */
     virtual size_t getNumKnots() const = 0;
 
+    /**
+     * Get the parametrization values corresponding to the knots of this spline
+     *
+     * So for example, if we had a 3-knot spline that is linearly interpolated, this
```

Comment:
It might be more clear to say that the knots are of equal 't' values from each other vs linearly interpolated.

it's clearer to me - @MathewMacDougall?

Reply:
Yea that's clear to me, but the current example also makes things pretty clear.

However "if we had a 3-knot spline that is linearly interpolated" makes it sound like the interpolation of the spline can vary and that determines the relationship of parameterization values to knots (ie. if it wasn't linearly interpolated, would the t-values be spread equally?).

Perhaps it would be more clear to say the parametrization values are spread linearly / equally without referencing the interpolation of the spline.

Reply:
> makes it sound like the interpolation of the spline can vary and that determines the relationship of parameterization values to knots (ie. if it wasn't linearly interpolated, would the t-values be spread equally?).

This is correct, it is implementation dependent. We could return `t={0, 0.25, 1.0}` and that would be perfectly acceptable. Added this comment:
```
     * NOTE: There is no guarantee that the knots will be at linearly interpolated 
     *       (equally spaced) intervals. This could return `{0, 0.25, 1.0}` and still be 
     *       well defined.
```